clang/test/dpct/kernel_without_name.cu

```diff
@@ -277,3 +277,28 @@ void run_foo7(T *a, const T *b, const T *c, const T *d, const T *e, const int f,
     foo_kernel6<<<grid, block, 0, stream>>>(a, b, c, d, e, f, g, h);
   }
 }
+
+template<class T1, class T2>
+struct UserStruct1 {
+  UserStruct1() {}
+  UserStruct1(UserStruct1 const &) {}
+};
+// CHECK: template <class T1, class T2> 
+// CHECK-NEXT: struct sycl::is_device_copyable<UserStruct1<T1, T2>> : std::true_type {};
+
+struct UserStruct2 {
+  UserStruct2() {}
+  UserStruct2(UserStruct2 const &) {}
+};
+// CHECK: template<>
+// CHECK-NEXT: struct sycl::is_device_copyable<UserStruct2> : std::true_type {};
+
+template<class V1, class V2>
+__global__ void foo_kernel7(UserStruct1<V1, V2>, UserStruct2) {}
```

Comment:
should we test nested type like: `UserStruct1<UserStruct2, V2>`

Reply:
test added